internal/oci/container_linux.go

```diff
@@ -31,50 +34,66 @@ func (c *Container) CleanupConmonCgroup(ctx context.Context) {
 	return c.seccompProfilePath
 }
 
-// getPidStartTime reads the kernel's /proc entry for stime for PID.
-// inspiration for this function came from https://github.com/containers/psgo/blob/master/internal/proc/stat.go
-// some credit goes to the psgo authors
-func getPidStartTime(pid int) (string, error) {
-	return GetPidStartTimeFromFile(fmt.Sprintf("/proc/%d/stat", pid))
+// GetPidStartTimeFromFile reads a file as if it were a /proc/<PID>/stat file,
+// looking for start time for PID. It is abstracted out to allow for unit testing.
+func GetPidStartTimeFromFile(file string) (string, error) {
+	_, startTime, err := getPidStatDataFromFile(file)
+	if err != nil {
+		return "", err
+	}
+
+	return startTime, nil
 }
 
-// GetPidStartTime reads a file as if it were a /proc/$pid/stat file, looking for stime for PID.
-// It is abstracted out to allow for unit testing
-func GetPidStartTimeFromFile(file string) (string, error) {
-	data, err := os.ReadFile(file)
+// getPidStatData returns the process state and start time for a given PID.
+func getPidStatData(pid int) (string, string, error) { //nolint:gocritic // Ignore unnamedResult.
+	return getPidStatDataFromFile(fmt.Sprintf(procStatFile, pid))
+}
+
+// getPidStatData parses the kernel's /proc/<PID>/stat file,
+// looking for the process state and start time for a given PID.
+func getPidStatDataFromFile(file string) (string, string, error) { //nolint:gocritic // Ignore unnamedResult.
+	f, err := os.Open(file)
 	if err != nil {
-		return "", fmt.Errorf("%v: %w", err, ErrNotFound)
+		return "", "", err
 	}
-	// The command (2nd field) can have spaces, but is wrapped in ()
-	// first, trim it
-	commEnd := bytes.LastIndexByte(data, ')')
-	if commEnd == -1 {
-		return "", fmt.Errorf("unable to find ')' in stat file: %w", ErrNotFound)
+	defer f.Close()
+
+	data, err := io.ReadAll(io.LimitReader(f, 4096))
+	if err != nil {
+		return "", "", err
 	}
 
-	// start on the space after the command
-	iter := commEnd + 1
-	// for the number of fields between command and stime, trim the beginning word
-	for field := 0; field < statStartTimeLocation-statCommField; field++ {
-		// trim from the beginning to the character after the last space
-		data = data[iter+1:]
-		// find the next space
-		iter = bytes.IndexByte(data, ' ')
-		if iter == -1 {
-			return "", fmt.Errorf("invalid number of entries found in stat file %s: %d: %w", file, field-1, ErrNotFound)
-		}
+	bracket := bytes.LastIndexByte(data, ')')
+	if bracket < 0 {
+		return "", "", fmt.Errorf("unable to find ')' in stat file: %w", ErrNotFound)
+	}
+
+	// Skip the process name and the white space after the right bracket.
+	fields := bytes.Fields(data[bracket+2:])
+
+	// Fields from the /proc/<PID>/stat file. See:
+	//   https://man7.org/linux/man-pages/man5/proc.5.html
+	//
+	// Field no. 3, the process state, such as "R", "S", "D", etc.
+	// Field no. 22, the process start time (clock ticks) since the system boot.
+	const startTimeFieldLocation = 22
```

Comment:
Not sure if this is a nit or a minuscule bug ..

Not that it would matter because the stat file is 52 fields long (50 when factoring out pid and name), but startTimeFieldLocation 22 is too high, given that you're looking into fields[19].

By explicitly naming the fields, the code would be more readable, and the offset / length error could be avoided:
```
const startTimeFieldLocation = 19
const stateFieldLocation = 0
(...)
if len(fields) < startTimeFieldLocation + 1 {
(...)
return string(fields[procFieldLocation]), string(fields[startTimeFieldLocation]), nil
```

Reply:
Surprisingly, the code originally looked like that, but then I decided to eliminate the variables to make it more concise.

Reply:
Done. Thank you!